File: sdks/go/test/getting_started_test.go
package snaptrade

import (
	"fmt"
	"math/rand"
	"os"
	"testing"
	"time"

	snaptrade "github.com/passiv/snaptrade-sdks/sdks/go"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func generateUUID() string {
	t := time.Now().UnixNano()
	rand.Seed(t)
	u := make([]byte, 16)
	rand.Read(u)
	u[8] = (u[8] | 0x40) & 0x7F
	u[6] = (u[6] & 0xF) | (4 << 4)
	return fmt.Sprintf("%x-%x-%x-%x-%x", u[0:4], u[4:6], u[6:8], u[8:10], u[10:])
}

func Test_snaptrade_GettingStarted(t *testing.T) {

	t.Run("Test ApiStatus", func(t *testing.T) {
		configuration := snaptrade.NewConfiguration()
		configuration.SetHost("http://127.0.0.1:4010")
		configuration.SetPartnerClientId("CLIENT_ID")
		client := snaptrade.NewAPIClient(configuration)

		request := client.APIStatusApi.Check()

		resp, httpRes, err := request.Execute()

		require.Nil(t, err)
		require.NotNil(t, resp)
		assert.Equal(t, 200, httpRes.StatusCode)
	})

	t.Run("Test GettingStarted", func(t *testing.T) {
		configuration := snaptrade.NewConfiguration()
		configuration.SetPartnerClientId(os.Getenv("SNAPTRADE_CLIENT_ID"))
		configuration.SetConsumerKey(os.Getenv("SNAPTRADE_CONSUMER_KEY"))
		client := snaptrade.NewAPIClient(configuration)

		// 1) Create a new user
		requestBody := snaptrade.NewSnapTradeRegisterUserRequestBody()
		userId := generateUUID()
		requestBody.SetUserId(userId)
		request := client.AuthenticationApi.RegisterSnapTradeUser(*requestBody)

		resp, httpRes, err := request.Execute()

		require.Nil(t, err)
		require.NotNil(t, resp)
		assert.Equal(t, 200, httpRes.StatusCode)

		// 2) Get user secret
		userSecret := resp.UserSecret
		require.NotEmpty(t, userSecret)

		// List users to ensure it was created
		listResp, _, _ := client.AuthenticationApi.ListSnapTradeUsers().Execute()
		require.Contains(t, listResp, userId)

		// 3) Get redirect URI
		loginResp, httpRes, err := client.AuthenticationApi.LoginSnapTradeUser(userId, *userSecret).Execute()
		require.Nil(t, err)
		require.NotNil(t, loginResp)
		assert.Equal(t, 200, httpRes.StatusCode)
		assert.NotEmpty(t, loginResp.LoginRedirectURI)

		// 4) Obtain account holdings data
		holdingsResp, httpRes, err := client.AccountInformationApi.GetAllUserHoldings(userId, *userSecret).Execute()
		require.Nil(t, err)
		require.NotNil(t, holdingsResp)
		assert.Equal(t, 200, httpRes.StatusCode)

		// 5) Delete the user
		deleteResp, httpRes, err := client.AuthenticationApi.DeleteSnapTradeUser(userId).Execute()
		require.Nil(t, err)
		require.NotNil(t, deleteResp)
		assert.Equal(t, 200, httpRes.StatusCode)
	})
}

File: sdks/php/test/GettingStartedTest.php
<?php

namespace SnapTrade\Test\Api;


use PHPUnit\Framework\TestCase;


class GettingStartedTest extends TestCase
{

    /**
     * Setup before running any test cases
     */
    public static function setUpBeforeClass(): void
    {
    }

    /**
     * Setup before running each test case
     */
    public function setUp(): void
    {
    }

    /**
     * Clean up after running each test case
     */
    public function tearDown(): void
    {
    }

    /**
     * Clean up after running all test cases
     */
    public static function tearDownAfterClass(): void
    {
    }

    /**
     * Test case for getting started flow
     *
     */
    public function testGettingStarted()
    {
        // 1) Initialize a client with your clientID and consumerKey.
        $snaptrade = new \SnapTrade\Client(
            clientId: getenv("SNAPTRADE_CLIENT_ID"),
            consumerKey: getenv("SNAPTRADE_CONSUMER_KEY")
        );

        // 2) Check that the client is able to make a request to the API server.
        $response = $snaptrade->apiStatus->check();
        print_r($response);

        // 3) Create a new user on SnapTrade
        $userId = (string)time();
        $userId .= "-php-sdk";
        $registerResponse = $snaptrade->authentication->registerSnapTradeUser($userId);
        $userSecret = $registerResponse->getUserSecret();

        // 4) Get a redirect URI. Users will need this to connect
        // their brokerage to the SnapTrade server.
        // {'broker': 'ALPACA', 'immediateRedirect': True, 'connectionPortalVersion': 'v2'}
        $redirectUri = $snaptrade->authentication->loginSnapTradeUser($userId, $userSecret, 'ALPACA', true, SENTINEL_VALUE, SENTINEL_VALUE, SENTINEL_VALUE, 'v2');
        print_r($redirectUri);

        // 5) Delete the user
        $deletedResponse = $snaptrade->authentication->deleteSnapTradeUser($userId);
        print_r($deletedResponse);
    }

    public function testSslConfig()
    {
        $snaptrade = new \SnapTrade\Client(
            clientId: getenv("SNAPTRADE_CLIENT_ID"),
            consumerKey: getenv("SNAPTRADE_CONSUMER_KEY"),
            host: "https://expired.badssl.com/",
            verifySsl: false,
        );
        $result = $snaptrade->apiStatus->check();
        print_r($result);
    }

    public function testGetUserAccountQuotes()
    {
        $snaptrade = new \SnapTrade\Client(
            clientId: getenv("SNAPTRADE_CLIENT_ID"),
            consumerKey: getenv("SNAPTRADE_CONSUMER_KEY"),
        );
        $user_id = getenv("SNAPTRADE_TEST_USER_ID");
        $user_secret = getenv("SNAPTRADE_TEST_USER_SECRET");
        $holdings = $snaptrade->accountInformation->getAllUserHoldings($user_id, $user_secret);
        $account_id = $holdings[0]->getAccount()->getId();
        $result = $snaptrade->trading->getUserAccountQuotes(
            user_id: $user_id,
            user_secret: $user_secret,
            account_id: $account_id,
            symbols: "AAPL",
            use_ticker: true
        );
        print_r($result);
        $this->assertTrue($result != null, "Response must not be null");
    }

    public function testGetUserAccountBalance()
    {
        $snaptrade = new \SnapTrade\Client(
            clientId: getenv("SNAPTRADE_CLIENT_ID"),
            consumerKey: getenv("SNAPTRADE_CONSUMER_KEY")
        );
        $userId = getenv("SNAPTRADE_TEST_USER_ID");
        $userSecret = getenv("SNAPTRADE_TEST_USER_SECRET");
        $accounts = $snaptrade->accountInformation->listUserAccounts(
            $userId,
            $userSecret
        );
        $response = $snaptrade->accountInformation->getUserAccountBalance($userId, $userSecret, $accounts[0]->getId());
        print_r($response);
    }

    /**
     * Test case for setting portfolio targets
     *
     */
    public function testSetPortfolioTargets()
    {
        $this->markTestSkipped("Can't use listAssetClasses to get asset id");

        // 1) Initialize a client with your clientID and consumerKey.
        $snaptrade = new \SnapTrade\Client(
            clientId: getenv("SNAPTRADE_CLIENT_ID"),
            consumerKey: getenv("SNAPTRADE_CONSUMER_KEY")
        );

        // 2) Check that the client is able to make a request to the API server.
        $response = $snaptrade->apiStatus->check();
        print_r($response);

        // 3) Create a new user on SnapTrade
        $userId = (string)time();
        $registerResponse = $snaptrade->authentication->registerSnapTradeUser($userId);
        $userSecret = $registerResponse->getUserSecret();

        // 4) Delete the user
        $deletedResponse = $snaptrade->authentication->deleteSnapTradeUser($userId);
        print_r($deletedResponse);
    }
}
